torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -283,15 +281,17 @@ struct GraphFuser {
       temporary_nodes.emplace_back(outer);
       auto inner_outputs = inner->outputs();
       auto outer_outputs = outer->outputs();
-      for (size_t i = 0; i < inner_outputs.size(); ++i)
+      for (size_t i = 0; i < inner_outputs.size(); ++i) {
         inner_to_outer[inner_outputs[i]] = outer_outputs[i];
+      }
     }
 
     // Replace uses of producer_group outputs and destroy the producer
     auto subgraph_outputs = producer_subgraph->outputs();
     for (size_t i = 0; i < subgraph_outputs.size(); ++i) {
       auto outer_output = inner_to_outer.at(subgraph_outputs[i]);
       producer_group->outputs()[i]->replaceAllUsesWith(outer_output);
+      aliasDb_->replaceWithNewValue(producer_group->outputs()[i], outer_output);
```

Comment:
Why are we doing this exactly? maybe add some comments to the new `replaceWithNewValue` invocations in this function

Reply:
The PR description says why we are doing this; we are basically rewiring the outputs to point to new outputs. I guess I can add a comment, although the comment at the beginning of the block sort of says that ?

Reply:
maybe just add something like: 
`new producer outputs have same aliasing properties as outer_output`